BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -137,9 +141,16 @@ uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
 void CAN1_RX0_IRQHandler(void)
 {
     CAN_Receive(CAN1, CAN_FIFO0, &RxMessage);
-
+    for(int i = 0; i < 8; i++){
+        rxFifo[rxFifoHead][i] = RxMessage.Data[i]; //store data in FIFO
+    }
+    (++rxFifoHead)%CAN_FIFO_SIZE; //increment FIFO Head
```

Comment:
Better to have this outside the loop, but it still doesn't do what you think it does.

Reply:
I think I fixed this